### PR TITLE
Added interceptor without spring data dependency

### PR DESCRIPTION
**Pull request by dekelpilli**
*Issue #, if available:*
https://github.com/aws/aws-xray-sdk-java/issues/45

*Description of changes:*
Added another abstract interceptor, extended by the existing one. This shouldn't break backwards-compatibility, but gives the option to users who don't have spring-data-jpa on their classpath.

Note that I haven't been able to test this yet, will be doing that soon.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Edit: I won't be able to test this, as the original library itself doesn't work for my use case. I'll leave this open in case it's still useful for this quite old issue.

### CODE REVIEW

**Comment by chanchiem** [comment]
Do you have any suggestions on how we can get this tested? 

Can we get some unit tests on this as well?

**Comment by dekelpilli** [comment]
> Do you have any suggestions on how we can get this tested?
> 
> Can we get some unit tests on this as well?

You can test this by simply doing what you would do with a normal Spring Boot app that uses this library, but make sure you don't have the Spring Data JPA library in the classpath. If the behaviour with BaseAbstractXRayInterceptor and no JPA lib is the same as BaseAbstractXRayInterceptor and the JPA lib present, then this change worked.

As for unit tests, I believe this would be non-trivial. Maybe an integration test with some mocked-out AWS-integration parts, but you would definitely need to load the Spring context to get a meaningful test. This is probably the reason that this module currently has no tests at all.

File: aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/BaseAbstractXRayInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+    }
+
+    protected abstract void xrayEnabledClasses();
+
+    @Pointcut("execution(* XRayTraced+.*(..))")
+    protected void xrayTracedClasses() {
+    }
+
+    protected void springRepositories() {
```
Comment (shengxil) [inline aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/BaseAbstractXRayInterceptor.java]: There is no need to leave the method in this base class as it intends to be Spring JBI free. We can move this and ```traceAroundRepositoryMethods()``` to the ```AbstractXRayInterceptor```

**Comment by shengxil** [comment]
Thanks for your contribution. I've verified the functionality but left a comment.

File: aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/BaseAbstractXRayInterceptor.java (hunk 1 above)
Comment (dekelpilli) [reply aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/BaseAbstractXRayInterceptor.java]: Good point, done.

**Pull request merged**